### PR TITLE
Substitute mock-accountant macro by a mock builder pallet

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

From the addition of this feature (https://github.com/foss3/runtime-pallet-library/pull/1), the mock accountant macro can be removed, and a mock builder pallet can be used instead.

## Changes and Descriptions

- Replace mock_accountant macro with a mock builder pallet.
- Replace `Rate` by `Quantity` (both represent the same type, but using `Quantity` for correctness).
  - Adapt consequently computed prices hardcoded values in tests
- Remove unused mocks from `cfg-test-utils`

### CODE REVIEW

**Comment by lemunozm** [comment]
Requires https://github.com/foss3/runtime-pallet-library/pull/15 to work

File: pallets/investments/src/tests.rs [hunk 1]
```diff
@@ -622,7 +622,7 @@ fn update_redeem_fails_when_collect_needed() {
 fn fulfillment_flow_for_everything_works() {
 	TestExternalitiesBuilder::build().execute_with(|| {
 		#[allow(non_snake_case)]
-		let PRICE: Rate = price_of(1, 2, 10);
+		let PRICE: Quantity = price_of(1, 2, 10);
```
Comment (mustermeiszer) [inline pallets/investments/src/tests.rs:625]: Why the change? In pallet pools we use `Rate`. It is confusing to change it here now IMO

**Review by mustermeiszer — COMMENTED** [review]
Besides the change to the `FixedPoint` type it looks good.

File: pallets/investments/src/tests.rs (hunk 1 above)
Comment (lemunozm) [reply pallets/investments/src/tests.rs:625]: This price is not "related" to the one used in `pallet-pool-system`. Actually, the own `pallet-investment` defines such type as `Quantity` under `T::BalanceRatio` (namely price). We do not want 27 decimals for them, we want 18. Basically I forgot to modify it when I updated all rates some months ago.

The rule we are using is prices: `Quantity`; rates (used by interests where decimals are quite significant): `Rate`.

File: libs/mocks/src/pools.rs [hunk 2]
```diff
@@ -64,6 +68,42 @@ pub mod pallet {
...
+		) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
+
+		#[allow(non_snake_case)]
+		pub fn mock_InvestmentAccountant_withdraw(
+			f: impl Fn(&T::AccountId, T::TrancheCurrency, T::Balance) -> DispatchResult + 'static,
+		) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
```
Comment (lemunozm) [inline libs/mocks/src/pools.rs:106]: The application of https://github.com/foss3/runtime-pallet-library/pull/16, @wischli 

Basically the pallet implements `deposit()` and `withdraw()` methods from two different traits. So we need to add `mock_<trait>_method()` to one of these `deposit()/withdraw()`.

In those cases, the `T::AccountId` used as generic for `InvestmentAccountant` was wrongly internally coded.

File: pallets/investments/src/tests.rs [hunk 3]
```diff
@@ -1289,11 +1289,11 @@ fn fulfillment_partially_works_low_price() {
 		{
 			assert_eq!(
 				free_balance_of(investment_account(INVESTMENT_0_0), AUSD_CURRENCY_ID),
-				207245508982035928145
+				207245508982035928140
```
Comment (lemunozm) [inline pallets/investments/src/tests.rs:1292]: Rounding issues due to modifying `Rate` (27 decimals) to `Quantity` (18 decimals).

File: pallets/investments/src/tests.rs [hunk 4]
```diff
@@ -2781,7 +2783,7 @@ fn collecting_fully_works() {
 				INVESTMENT_0_0
 			));
 			assert_eq!(
-				n_last_event(3),
+				n_last_event(4),
```
Comment (lemunozm) [inline pallets/investments/src/tests.rs:2786]: The expected event is behind another produced event of killing an account.

I think the rounding issues put the account under the existential deposit value, and the account is killed.

File: pallets/investments/src/benchmarking.rs [hunk 5]
```diff
@@ -56,6 +56,9 @@ mod benchmarks {
 
 	#[benchmark]
 	fn update_invest_order() -> Result<(), BenchmarkError> {
+		#[cfg(test)]
+		crate::mock::configure_accountant_mock();
```
Comment (lemunozm) [inline pallets/investments/src/benchmarking.rs:60]: Completely astonished by the needed for this.

Without the above line, the test crashes because the mock closure methods are not defined. Nevertheless, `configure_accountant_mock()` is already called under `TextExternatilitiesBuilder::build()`, so I would expect to have already defined the mock methods. 

It seems like for benchmarking, somehow, the thread used to build the externality is not the same as the one used to benchmark. And then, the internal `thread_local` used by the mock builder doesn't correspond to the one used.

Comment (lemunozm) [reply pallets/investments/src/benchmarking.rs:60]: Actually, for testing the 4 benchmarking, just one call to `TextExternatilitiesBuilder::build()` is done

File: pallets/investments/src/tests.rs (hunk 1 above)
Comment (mustermeiszer) [reply pallets/investments/src/tests.rs:625]: Ahh, I see. Sorry, my bad you are right of course. I did overlook that the tranches are using `Quantity` too.

File: pallets/investments/src/benchmarking.rs (hunk 5 above)
Comment (mustermeiszer) [reply pallets/investments/src/benchmarking.rs:65]: Interesting finding.

**Review by mustermeiszer — APPROVED** [review]
Thanks for the clarification!

**Review by mustermeiszer — APPROVED** [review]
